docs/guide/extending/custom_fieldtype_comparison.md

```diff
@@ -0,0 +1,117 @@
+# Creating custom version comparison of Field Types
+
+In the Back Office, you can compare the contents of a Field Type.
+Out of the box, you can compare versions of the TextLine Field Type (`ezstring`).
+Comparing is possible only between two versions of the same Field Type that are in the same language.
+
+You can add the possibility to compare other Field Types, including the custom ones.
+You can base the configuration on the comparison mechanism created for the `ezstring` Field Type.
+
+## Field Type configuration
+
+First, ensure that the Field Type to compare implements the `EzSystems\EzPlatformContentComparison\FieldType\Comparable` interface with the following method:
+
+``` php
+use eZ\Publish\SPI\FieldType\Value;
+use EzSystems\EzPlatformContentComparison\Comparison\ComparisonData;
+
+interface Comparable
+{
+    public function getDataToCompare(Value $value): ComparisonData;
+}
+```
+
+This method fetches the data to compare and determines which [comparison engines](#comparison-engine) should be used.
+Therefore, the `ComparisonData` object is specific to the Field Type you want to compare.
+
+In case of `ezstring`, the implementation is:
+
+``` php
+public function getDataToCompare(Value $value): ComparisonData;
+{
+    return new TextLine([
+        'textLine' => new StringComparisonValue([
+            'value' => $value->text,,
+        ]),
+    ]);
+}
+```
+
+Also, the Field Type requires additional configuration (e.g. `comparable_fieldtypes.yaml`).
+An exemplary configuration, in this case for `ezstring`, looks the following way:
+
+``` yaml
+EzSystems\EzPlatformContentComparison\FieldType\TextLine\Comparable:
+  tags:
+    - { name: ezplatform.field_type.comparable, alias: ezstring }
+```
+
+## Comparison engine
+
+The comparison engine handles the operations required for comparing the contents of Field Types.
+Note that each Field Type requires a separate comparison engine.
+
+When creating a custom engine, ensure that it implements the `EzSystems\EzPlatformContentComparison\Engine` interface:
+
+``` php
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformContentComparison\Engine;
+
+use EzSystems\EzPlatformContentComparison\Comparison\ComparisonData;
+```
+
+The engine requires also configuration similar to the following one:
+
+``` yaml
+EzSystems\EzPlatformContentComparison\Engine\Value\StringValueComparisonEngine: ~
```

Comment:
only this is needed for registering engine, in your example you registered 3 of them.

```
EzSystems\EzPlatformContentComparison\Engine\FieldType\TextLineComparisonEngine:
  tags:
    - { name: ezplatform.field_type.comparable.engine, supported_type: EzSystems\EzPlatformContentComparison\Comparison\Field\TextLine }
```

Reply:
Rest of them comes out of the box, and handles cases you mentioned, but there is no need to redefine them, unless you want to override them, but then we are on symfony container grounds.